lib/go-atscfg/remapdotconfig.go

```diff
@@ -1002,11 +1002,53 @@ func serverIsLastCacheForDS(server *Server, ds *DeliveryService, topologies map[
 		return topoPlacement.IsLastCacheTier, nil
 	}
 
-	return noTopologyServerIsLastCacheForDS(server, ds), nil
+	return noTopologyServerIsLastCacheForDS(server, ds, cacheGroups), nil
 }
 
 // noTopologyServerIsLastCacheForDS returns whether the server is the last tier for the DS, if the DS has no Topology.
 // This helper MUST NOT be called if the DS has a Topology. It does not check.
-func noTopologyServerIsLastCacheForDS(server *Server, ds *DeliveryService) bool {
-	return strings.HasPrefix(server.Type, tc.MidTypePrefix) || !ds.Type.UsesMidCache()
+func noTopologyServerIsLastCacheForDS(server *Server, ds *DeliveryService, cgs map[tc.CacheGroupName]tc.CacheGroupNullable) bool {
+	if strings.HasPrefix(server.Type, tc.MidTypePrefix) {
+		return true // if the type is "MID" it's always the last cache for non-topologies
+	}
+	if !ds.Type.UsesMidCache() {
+		return true // if this DS type never uses mids (for pre-topology type-parentage), it's the last cache
+	}
+
+	// pre-topology parentage is based on Cachegroups
+
+	if server.Cachegroup == nil {
+		// if the server has no CG (which TO shouldn't allow), it can't possibly have parents.
+	}
```

Comment:
Is this *if* statement supposed to be empty?

Reply:
Nope, missed the return. Fixed.